sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -128,6 +128,15 @@ object ScalaReflection extends ScalaReflection {
     case _ => false
   }
 
+  def isValueClass(tpe: `Type`): Boolean = {
+    tpe.typeSymbol.asClass.isDerivedValueClass
+  }
+
+  /** Returns the name and type of the underlying parameter of value class `tpe`. */
+  def getUnderlyingParameterOf(tpe: `Type`): (String, Type) = {
+    getConstructorParameters(tpe).head
```

Comment:
is there a more official way to get the value class field name?

Reply:
not sure, I can't find any